qiskit/providers/backend_compat.py

```diff
@@ -13,185 +13,256 @@
+                "Please add new gate class to Qiskit or provide GateConfig for this name.",
+                name,
+            )
+            unsupported_instructions.append(name)
+
+    for name in unsupported_instructions:
+        all_instructions.remove(name)
+
+    # Create empty inst properties from gate configs
+    for name, spec in gate_configs.items():
+        if hasattr(spec, "coupling_map"):
+            coupling_map = spec.coupling_map
+            prop_name_map[name] = dict.fromkeys(map(tuple, coupling_map))
+        else:
+            prop_name_map[name] = None
+
+    # Populate instruction properties
+    if properties:
+        qubit_properties = [
+            QubitProperties(
+                t1=properties.qubit_property(qubit_idx)["T1"][0],
+                t2=properties.qubit_property(qubit_idx)["T2"][0],
+                frequency=properties.qubit_property(qubit_idx)["frequency"][0],
+            )
+            for qubit_idx in range(0, configuration.num_qubits)
+        ]
+
+        in_data["qubit_properties"] = qubit_properties
 
-    # Standard gates library mapping, multicontrolled gates not included since they're
-    # variable width
-    name_mapping = get_standard_gate_name_mapping()
-    target = None
-    if custom_name_mapping is not None:
-        name_mapping.update(custom_name_mapping)
-    faulty_qubits = set()
-    # Parse from properties if it exsits
-    if properties is not None:
         if filter_faulty:
-            faulty_qubits = set(properties.faulty_qubits())
-        qubit_properties = qubit_props_list_from_props(properties=properties)
-        target = Target(
-            num_qubits=configuration.n_qubits,
-            qubit_properties=qubit_properties,
-            concurrent_measurements=getattr(configuration, "meas_map", None),
-        )
-        # Parse instructions
-        gates: Dict[str, Any] = {}
-        for gate in properties.gates:
-            name = gate.gate
-            if name in name_mapping:
-                if name not in gates:
-                    gates[name] = {}
-            else:
-                raise QiskitError(
-                    f"Operation name {name} does not have a known mapping. Use "
-                    "custom_name_mapping to map this name to an Operation object"
-                )
-
-            qubits = tuple(gate.qubits)
-            if filter_faulty:
-                if any(not properties.is_qubit_operational(qubit) for qubit in qubits):
+            faulty_qubits = properties.faulty_qubits()
+
+        for name in prop_name_map.keys():
+            for qubits, params in properties.gate_property(name).items():
+                in_param = {
+                    "error": params["gate_error"][0] if "gate_error" in params else None,
+                    "duration": params["gate_length"][0] if "gate_length" in params else None,
+                }
+                inst_prop = InstructionProperties(**in_param)
```

Comment:
Is there a reason you need to use an intermediate dictionary here? Wouldn't it create one less object if you just did:

```suggestion
error = params["gate_error"][0] if "gate_error" in params else None
duration = params["gate_length"][0] if "gate_length" in params else None
inst_prop = InstructionProperties(error=error, duration=duration)
```

Reply:
yeah this is copied from provider which I wrote where I needed to apply schema. This code is not really inefficient as you say.